lib/route/src/Obelisk/Route.hs

```diff
@@ -825,14 +832,22 @@ _R
   -> Prism' (R f) a
 _R variant = dSumGEqPrism variant . iso runIdentity Identity
 
+{-# DEPRECATED prismEncoder "Use reviewEncoder instead" #-}
 -- | An encoder that only works on the items available via the prism. An error will be thrown in the parse monad
 -- if the prism doesn't match.
+-- This encoder is contravariant like 'review': @prismEncoder (f . g) = prismEncoder g . prismEncoder f@
```

Comment:
Let's down-level the language here.  Why rely on the user understanding the concept of contravariance?

> `Note that a 'Prism' from @a@ to @b@ will produce an 'Encoder' from @b@ to @a@ (i.e. 'prismEncoder' is a contravariant functor from the category of prisms to the category of encoders), just like `review` produces a function @b -> a@.  This is because 'Prism's extract values, in a way that might fail, in their forward direction and inject values, in a way that cannot fail, in their reverse direction; whereas Encoders encode, which cannot fail, in their forward direction, and decode, which can fail, in their reverse direction.`

This way, we're not relying on the user already knowing the terminology, but we are teaching it to them.  Also, we're explaining the intuition of *why* this is the case.

Reply:
This is a good idea. I was going to address this later when we actually did a full-scale attack on the docs of this module in general, but no harm in starting now.